server/src/main/java/org/elasticsearch/index/codec/vectors/cluster/KMeansIntermediate.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.index.codec.vectors.cluster;
+
+import org.apache.lucene.util.hnsw.IntToIntFunction;
+
+/**
+ * Intermediate object for clustering (partitioning) a set of vectors
+ */
+public class KMeansIntermediate extends KMeansResult {
```

Comment:
package private?

basically, I think most (if not all) the classes here should be package private.